### PR TITLE
[ci] Adding tox

### PR DESCRIPTION
**Pull request by john-bodley**
This PR adds [tox](https://tox.readthedocs.io/en/latest/) a common Python testing environment. Note I also added a `requirements.txt` file (used in `tox`) which freezes all the dependencies (including the extras) which means we're no longer skipping certain tests or mocking the Tornado logic. 

In future PRs I plan to:

- Replace `flake8` with `black`, `isort`, and `pylint`. 
- Add typing (with enforcement) via `mypy`. 

to: @betodealmeida @mistercrunch

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-e .[async,cli,pandas,sqlalchemy]
+pandas<0.21
```
Comment (john-bodley) [inline requirements.in]: Most of these requirements are to satisfy Python 3.4 which is no longer supported.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,17 @@
 language: python
-python:
-  - "2.7"
-  - "3.4"
-  - "3.6"
+matrix:
+  include:
+    - env: TOXENV=flake8
+    - env: TOXENV=py27
```
Comment (mistercrunch) [inline .travis.yml]: deprecating py27!

**Comment by MSadeghzadehG** [comment]
I can't see how these changes could resolve missing tabulate module problem?

**Comment by john-bodley** [comment]
@MSadeghzadehG if you look in `requirements.txt` (which in autogenerated from `requirements.in` using `pip-compile`) you can see a line item for the `tabulate` package.

**Comment by MSadeghzadehG** [comment]
@john-bodley Nice.

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,3 @@
+-e .[async,cli,pandas,sqlalchemy]
+pandas<1.0.0
```
Comment (mistercrunch) [inline requirements.in:2]: why dup pandas here? isn't it included in `.[pandas]`?

Comment (john-bodley) [reply requirements.in:2]: @mistercrunch let me look into it. The constraint should be in `setup.py` if possible (assuming that `pip-compile` can handle this). I know that extra requirements are challenging for `pip-compile` (hence the reason for the addition of the `requirements.in` file).

**Comment by john-bodley** [comment]
@mistercrunch I addressed your comment.

**Comment by SongYunSeop** [comment]
Is it still working in progress?

**Comment by john-bodley** [comment]
@SongYunSeop this PR is ready for review.

**Comment by mistercrunch** [comment]
neato

**Pull request merged**